### PR TITLE
Wildcard: simplify Tabs implementation 

### PR DESCRIPTION
**Pull request by oleggromov**
## Why

While doing migration to the new component (https://github.com/sourcegraph/sourcegraph/pull/29920), I came across a bug that I couldn't fix and even understand fully. This all is due to the overly sophisticated implementation of Tabs with a controversial internal state, which pre-rendered and cached panels with respect to `lazy` and `behavior`, and some effects, which may have been in a race condition once in a while.

Since the implementation was obviously more complex than it could be, I decided to simplify it. 

## What's changed
Instead of saving children in internal state, which is passed around as context, we only save two required things:
1. `settings`, which are `Tabs`'s props - to be used down the component tree
2.  and `index` of each rendered panel, which is required to decide `shouldPanelRender` or not

Everything else wasn't really needed.

### Behavior
To re-iterate, `Tabs` support 3 types of behavior:
1. `lazy={false}` - Just render each panel no matter which one is selected
2. `lazy={true} behavior='forceRender'` - Only render currently selected panel
3. `lazy={true} behavior='memoize'` - Render each panel once and don't _un-render_ once selected index is changed

The `3rd` case now has a slightly modified (yet important) behavior. If children of a panel change when it's not selected, it will be un-rendered and re-rendered when it's selected back. Before it wasn't the case and could lead to strange and hard to understand bugs.

### CODE REVIEW

File: client/wildcard/src/components/Tabs/Tabs.tsx [hunk 1]
```diff
@@ -8,34 +8,46 @@ import {
...
-import { useTabPanelsState } from './useTabPanelsState'
-import { TabsState, useTabs } from './useTabs'
+import { useShouldPanelRender } from './useShouldPanelRender'
 
-interface TabsProps extends ReachTabsProps, TabsState {
+export interface TabsSettings {
+    /* Tab component font size.
+     * Default is "small" */
+    size?: 'small' | 'medium' | 'large'
+    /* true: only load the initial tab when tab component mounts
```
Comment (umpox) [inline client/wildcard/src/components/Tabs/Tabs.tsx:24]:
```suggestion
    /** Tab component font size.
     * Default is "small" */
    size?: 'small' | 'medium' | 'large'
    /** true: only load the initial tab when tab component mounts
```

So it is valid JSDoc and shows in hover definitions :)

**Review by umpox — APPROVED** [review]
Thanks! Looks like a nasty issue to debug

File: client/wildcard/src/components/Tabs/Tabs.tsx [hunk 2]
```diff
@@ -45,20 +57,21 @@ export type { TabsProps, TabPanelsProps, TabPanelProps }
  * See: https://reach.tech/tabs/
  *
  */
-export const Tabs: React.FunctionComponent<TabsProps> = props => {
+export const Tabs: React.FC<TabsProps> = props => {
```
[the remark this answers is not in the document]
Comment (oleggromov) [reply client/wildcard/src/components/Tabs/Tabs.tsx]: Done.

File: client/wildcard/src/components/Tabs/Tabs.tsx (hunk 1 above)
Comment (oleggromov) [reply client/wildcard/src/components/Tabs/Tabs.tsx]: That's a great catch, thanks Tom!

**Pull request merged**